### PR TITLE
[hal] more succinctly expressed by calling .find(p) instead.

### PR DESCRIPTION
**Pull request by Michael-Lfx**
Fixes #issue
PR checklist:
- [x] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends: metal
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: src/hal/src/adapter.rs [hunk 1]
```diff
@@ -187,12 +187,11 @@ impl<B: Backend> Adapter<B> {
         let requested_family = self
             .queue_families
             .iter()
-            .filter(|family| {
+            .find(|family| {
                 C::supported_by(family.queue_type())
                     && selector(&family)
```
Comment (kvark) [inline src/hal/src/adapter.rs]: nit: I don't think we need `&` before `familty` here now

**Comment by Michael-Lfx** [comment]
done.  @kvark

**Comment by kvark** [comment]
@Michael-Lfx is it? I don't see it

**Comment by Michael-Lfx** [comment]
> @Michael-Lfx is it? I don't see it

Sorry. I forget to force push it.

**Comment by kvark** [comment]
Bors r+

**Pull request merged**